package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -565,6 +569,19 @@ def _format_hint(thing):
         # format='H5MD' is used
         return HAS_H5PY and isinstance(thing, h5py.File)
 
+    @staticmethod
+    def parse_n_atoms(filename):
+        with h5py.File(filename, 'r') as f:
+            for group in f['particles/trajectory']:
+                if group in ('position', 'velocity', 'force'):
+                    n_atoms = f[f'particles/trajectory/{group}/value'].shape[1]
+                    return n_atoms
+
+            raise NoDataError("Could not construct minimal topology from the "
```

Comment:
It's syntactically the same, but isn't this an example of where a for/else loop is a cool idea?

Reply:
sure but pylint hates it for some reason